### PR TITLE
Ability to patch ruby source code

### PR DESCRIPTION
**Pull request by hone**

### CODE REVIEW

File: build.rb [hunk 1]
```diff
@@ -17,14 +17,18 @@ def pipe(command)
   output
 end
 
-def fetch(url)
+def fetch(url, name = nil)
```
Comment (schneems) [inline build.rb]: Would be better as a named argument, also maybe `patch_name` instead of `name`.

**Review by schneems — CHANGES_REQUESTED** [review]
Minor comment on naming

File: build.rb (hunk 1 above)
Comment (hone) [reply build.rb]: Well, `name` is used here b/c in theory you could name other stuff since it's not just specific to patches.

Comment (hone) [reply build.rb]: In theory, using keyword argument means we can't built cedar binaries, which is probably ok.

Comment (schneems) [reply build.rb]: ya, cedar is read only mode.

Instead of name, maybe `file_name`.

**Comment by schneems** [comment]
Lets' merge as is

**Pull request merged**